subworkflows/local/shortread_fastp.nf

```diff
@@ -0,0 +1,61 @@
+/*
+Process short raw reads with FastP
+*/
+
+include { FASTP as FASTP_SINGLE       } from '../../modules/nf-core/modules/fastp/main'
+include { FASTP as FASTP_PAIRED       } from '../../modules/nf-core/modules/fastp/main'
+
+workflow SHORTREAD_FASTP {
+    take:
+    reads // [[meta], [reads]]
+
+    main:
+    ch_versions = Channel.empty()
+    ch_multiqc_files      = Channel.empty()
+
+    ch_input_for_fastp = reads
+                            .dump(tag: "pre-fastp_branch")
+                            .branch{
+                                single: it[0]['single_end'] == true
+                                paired: it[0]['single_end'] == false
+                            }
```

Comment:
Why do you branch the channel here when the fastp module is perfectly capable of handling both single- and paired-end reads?

Reply:
Because if we merge paired end reads, we have to pull out a separte channel (`.merged` rather than just `.reads`)